### DISCUSSION TITLE
Over the air Update

### DISCUSSION THREAD

**Discussion by Shivaraj199** [polarfire-soc/polarfire-soc-documentation#386] [How to]
is there any a way for software Update over the AIR.
I architecture states the HSS always picks up  the PAyload from sector0 or the GPT partions. But how to modify HSS for Software Udpate, do i need to change the hss_boot_service.c?
Is there any example or documentation for Over the AIR software Update

**Comment by vfalanis** [comment]
Hello @Shivaraj199 ,

Are you looking to update the HSS at runtime? Or what is your use case?

One thing you could do is to use the "Auto Update" feature to update the HSS. There is a document explaining how to use auto update as well as how to generate a bitstream that supports this feature. When exporting the bitstream you could generate a bitstream that contains the HSS only.

More information on auto update can be found here:

https://github.com/polarfire-soc/polarfire-soc-documentation/blob/master/how-to/re-programming-the-fpga-from-linux.md
